scala-thoughts.md

```diff
@@ -0,0 +1,55 @@
+## Scala thoughts
+
+#### Likes
+
+Overall, my impression of Scala is that it is very intuitive.
+To me, it seems most similar to Python, but with static typing.
+
+I personally like the static typing because it can make the code a lot more
+readable and make it easier to understand what the input and output of each
+method is.
+
+The code seems a lot more concise than a lot of other languages I've worked
+with.
+For example, the `if` `then` `else` syntax only has to have those keywords
+without any braces, parentheses, or colons.
+Another example is inserting values into strings just requires `s` before and
+`$` in the string making it very readable.
+
+A bit of a minor detail, but the I enjoy the syntax for `for` loops.
+The arrow is a little strange, but just stating the iteration variable name and
+then min `to` max seems so much more intuitive to me than using
+`range(min, max)` or `i = min; i < max; i++`.
+
+#### Dislikes
+
+One thing I noticed was a bit difficult to do in Scala was creating blank lists.
+I wanted to define a list with a specific type and size that I then update
+values of in a loop, but I couldn't figure out how to initialize the list.
+
+Another thing I found annoying was the lack of `break` statements.
```

Comment:
Wow, I was like "there's no way break statements aren't built-in", but you're right. Apparently there are packages that can get it done (https://www.geeksforgeeks.org/break-statement-in-scala/) but it's definitely a silly omission.